consensus/src/dag/bullshark.rs

```diff
@@ -97,61 +106,56 @@ impl Bullshark {
         reachable_nodes.keys().contains(&target.digest())
     }
 
-    fn order_anchors(&mut self, anchor: Node) -> Vec<Node> {
-        let mut anchor_stack = Vec::new();
-        let mut round = anchor.round();
-        assert_eq!(round % 2, 0);
-        let mut wave = anchor.round() / 2;
-        let new_ordered_wave = wave;
+    /// Given an anchor that's committed with f+1, find the lowest ancestor anchor
+    /// and update the leader reputation
+    fn find_first_anchor_to_commit(&mut self, mut current_anchor: Node) -> Node {
+        let mut current_round = current_anchor.round();
 
-        let mut current_anchor = anchor;
+        while current_round > self.lowest_unordered_anchor_round {
+            current_round -= 2;
+            let prev_anchor_peer_id = self
+                .proposer_election
+                .get_round_anchor_peer_id(current_round);
 
-        // wave -= 1;
-        // round -= 2;
-
-        assert!(wave >= self.lowest_unordered_anchor_wave);
-        while wave > self.lowest_unordered_anchor_wave {
-            wave -= 1;
-            round -= 2;
-            let prev_anchor_peer_id = self.proposer_election.get_round_anchor_peer_id(wave);
-
-            let is_path =
-                if let Some(prev_anchor) = self.dag[round as usize].get(&prev_anchor_peer_id) {
+            if self.dag[current_round as usize]
+                .get(&prev_anchor_peer_id)
+                .map_or(false, |prev_anchor| {
                     self.strong_path(&current_anchor, prev_anchor)
-                } else {
-                    false
-                };
-
-            if is_path {
-                anchor_stack.push(std::mem::replace(
-                    &mut current_anchor,
-                    self.dag[round as usize]
-                        .remove(&prev_anchor_peer_id)
-                        .unwrap(),
-                ));
+                })
+            {
+                current_anchor = self.dag[current_round as usize]
+                    .get(&prev_anchor_peer_id)
+                    .unwrap()
+                    .clone();
             }
-
-            // wave -= 1;
-            // round -= 2;
         }
 
-        anchor_stack.push(current_anchor);
-        self.lowest_unordered_anchor_wave = new_ordered_wave + 1;
-        anchor_stack
-    }
+        let anchor_round = current_anchor.round();
```

Comment:
I think `current_anchor` here should be the original one passed to the function, but this one uses the reset one in line 126.